### ISSUE TITLE
Invalid message id may crash worker

### ISSUE THREAD

**Issue by kilgor** [slact/nchan#398]
```
2017/10/02 20:58:40 [alert] 30605#30605: worker process 30606 exited on signal 6 (core dumped)
ker process: /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/memory/memstore.c:2018: chanhead_find_next_message: Assertion `msgid->tagcount == 1 && first->msg->id.tagcount == 1' failed.
```

**Comment by kilgor** [comment]
```
#0  0x00007f9c9e57c067 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007f9c9e57d448 in __GI_abort () at abort.c:89
#2  0x00007f9c9e575266 in __assert_fail_base (fmt=0x7f9c9e6adf18 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=assertion@entry=0x7f9c9d655580 "msgid->tagcount == 1 && first->msg->id.tagcount == 1",
    file=file@entry=0x7f9c9d654330 "/usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/memory/memstore.c", line=line@entry=2018,
    function=function@entry=0x7f9c9d6563e0 <__PRETTY_FUNCTION__.29160> "chanhead_find_next_message") at assert.c:92
#3  0x00007f9c9e575312 in __GI___assert_fail (assertion=assertion@entry=0x7f9c9d655580 "msgid->tagcount == 1 && first->msg->id.tagcount == 1",
    file=file@entry=0x7f9c9d654330 "/usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/memory/memstore.c", line=line@entry=2018,
    function=function@entry=0x7f9c9d6563e0 <__PRETTY_FUNCTION__.29160> "chanhead_find_next_message") at assert.c:101
#4  0x00007f9c9d633a8c in chanhead_find_next_message (ch=0x1c75560, msgid=msgid@entry=0x1da21b0, status=status@entry=0x7ffc3ac7b5e4)
    at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/memory/memstore.c:2018
#5  0x00007f9c9d633cf2 in nchan_store_async_get_message (channel_id=0x1c75560, msg_id=0x269cbd8, cf=<optimized out>, callback=0x7f9c9d627530 <spool_fetch_msg_callback>, privdata=0x269ccf0)
    at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/memory/memstore.c:2757
#6  0x00007f9c9d62688d in spool_fetch_msg (spool=0x269cbd8) at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/spool.c:475
#7  0x00007f9c9d62725b in spooler_add_subscriber (self=0x1c755c8, sub=0x2699210) at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/spool.c:691
#8  0x00007f9c9d632858 in nchan_store_subscribe_continued (channel_status=<optimized out>, d=0x27e65e0, _=<optimized out>) at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/store/memory/memstore.c:2391
#9  0x00007f9c9d61ba3a in nchan_subscriber_subscribe (sub=0x2699210, ch_id=<optimized out>) at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/subscribers/common.c:310
#10 0x00007f9c9d606876 in nchan_pubsub_handler (r=0x269abe0) at /usr/local/src/nginx/build/nginx-1.12.1/debian/modules/nchan-nginx-module/src/nchan_module.c:605
#11 0x0000000000450f4f in ngx_http_core_content_phase ()
#12 0x000000000044b365 in ngx_http_core_run_phases ()
#13 0x00000000004575f3 in ngx_http_process_request ()
#14 0x0000000000457fce in ?? ()
#15 0x000000000043df37 in ?? ()
#16 0x00000000004335d7 in ngx_process_events_and_timers ()
#17 0x000000000043b7e5 in ?? ()
#18 0x000000000043a20a in ngx_spawn_process ()
#19 0x000000000043cd4e in ngx_master_process_cycle ()
#20 0x0000000000411e1d in main ()
```

**Comment by slact** [comment]
Nchan version? Nginx config?

**Comment by kilgor** [comment]
Can you provide me public ssh key? I gave you access to VPS where we test nchan solution

**Comment by slact** [comment]
```
ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIOlzTl9tCVC/I8cE6TwompznQ/GzRnefs+VViJDXZtrP
```

**Comment by kilgor** [comment]
I send you email ;)

**Comment by slact** [comment]
Fixed in 954edb1f. The issue was bad handling of incorrect message ids. One of your clients was sending a multichannel id for a non-multiplexed channel. This was handled incorrectly, hence the worker crash.
